src/components/InputBook.jsx

```diff
@@ -1,26 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
 
-const InputBook = () => (
-  <div className="form-container">
-    <h3 className="form-heading">Add a new book</h3>
-    <form>
-      <input
-        className="title"
-        name="title"
-        type="text"
-        placeholder="Book Title"
-        required
-      />
-      <input
-        className="author"
-        name="author"
-        type="text"
-        placeholder="Author"
-        required
-      />
-      <button type="button" className="add-btn">Add Book</button>
-    </form>
-  </div>
-);
+const InputBook = () => {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const dispatch = useDispatch();
+  const books = useSelector((state) => state.books);
+
+  const formSubmit = (e) => {
+    e.preventDefault();
+    if (title && author) {
+      dispatch(addBook({
+        id: books.length + 1,
+        title,
+        author,
+      }));
+      setTitle('');
+      setAuthor('');
+    }
+  };
+
+  return (
+    <div className="form-container">
+      <h3 className="form-heading">Add a new book</h3>
+      <form onSubmit={formSubmit}>
+        <input
+          className="title"
+          name="title"
+          type="text"
+          placeholder="Book Title"
+          required
+        />
+        <input
+          className="author"
+          name="author"
+          type="text"
+          placeholder="Author"
+          required
+        />
+        <button type="button" className="add-btn">
+          Add Book
+        </button>
+      </form>
```

Comment:
- [ ] Currently, book data from the form is not being added to the boos array in your store. You are correctly fetching the data from the store but are not posting it. You should use the value prop on the input fields and make use of the `onChange` prop then change the being pushed to be the value entered on the input fields. You are required to add a book and that book is displayed on the fly. Please work on that 👍 
- [ ] See the Requirement
![image](https://user-images.githubusercontent.com/50941074/174143177-e0f70a36-d59b-46ef-8ab9-d571ed1d2376.png)


**Hint:**
you could have these properties passed to your input fields and handle the posting
```mdx
value={title}
onChange={(e) => setTitle(e.target.value)}
```
and similar to the author input 👍

Reply:
@AmaduKamara thank you for pointing this out, I have added this in my form and now it's working as intended.